libs/fluid-elements/button/src/lib/button.spec.ts

```diff
@@ -0,0 +1,253 @@
+      );
+      expect(classList).toContain('fluid-emphasis--high');
+      fixture.removeAttribute('emphasis');
+      await tick();
+      classList = Array.from(
+        fixture.shadowRoot?.querySelector('button')?.classList!,
+      );
+      expect(classList).toContain('fluid-emphasis--medium');
+    });
+
+    it('should fall back to the default when setting the property to null ', async () => {
+      fixture.emphasis = 'high';
+      await tick();
+      let classList = Array.from(
+        fixture.shadowRoot?.querySelector('button')?.classList!,
+      );
+      expect(classList).toContain('fluid-emphasis--high');
+      fixture.emphasis = null;
+      await tick();
+      classList = Array.from(
+        fixture.shadowRoot?.querySelector('button')?.classList!,
+      );
+      expect(classList).toContain('fluid-emphasis--medium');
+    });
+  });
+
+  describe('disabled attribute', () => {
+    it('should set the disabled state when the attribute is present', async () => {
+      fixture.setAttribute('disabled', '');
+      await tick();
+      const shadowedButton = fixture.shadowRoot?.querySelector('button')!;
+      expect(shadowedButton.hasAttribute('disabled')).toBe(true);
+    });
+
+    it('should set the disabled state when the attribute is set', async () => {
+      fixture.setAttribute('disabled', 'disabled');
+      await tick();
+      const shadowedButton = fixture.shadowRoot?.querySelector('button')!;
+      expect(shadowedButton.hasAttribute('disabled')).toBe(true);
+    });
+
+    it('should set the disabled state when the property is set', async () => {
+      fixture.disabled = true;
+      await tick();
+      const shadowedButton = fixture.shadowRoot?.querySelector('button')!;
+      expect(shadowedButton.hasAttribute('disabled')).toBe(true);
+    });
+
+    it('should reset the disabled state when the attribute is removed', async () => {
+      fixture.setAttribute('disabled', '');
+      await tick();
+      fixture.removeAttribute('disabled');
+      await tick();
+      const shadowedButton = fixture.shadowRoot?.querySelector('button')!;
+      expect(shadowedButton.hasAttribute('disabled')).toBe(false);
+    });
+
+    it('should reflect the disabled attribute to the host', async () => {
+      fixture.disabled = true;
+      await tick();
+      expect(fixture.hasAttribute('disabled')).toBeTruthy();
+    });
+  });
+
+  describe('click handler', () => {
+    let mockHandler;
+    beforeEach(() => {
+      mockHandler = jest.fn();
+      fixture.addEventListener('click', mockHandler);
+    });
+    it('should call the handler when clicked', async () => {
+      fixture.click();
+      expect(mockHandler).toHaveBeenCalled();
+    });
+
+    // TODO: Figure out how we can disable click interaction
+    // without the use of css.
+    // It works in the browser for native clicks, because the pointer-events styling
+    // kicks in. But in the tests it's somehow hard to reproduce
```

Comment:
would this be something for e2e tests?

Reply:
quite possibly, yes. For an enhancement of the webcomponents library certainly.